src/System.Private.CoreLib/shared/System/Buffers/Text/Utf8Parser/Utf8Parser.Date.J.cs

```diff
@@ -0,0 +1,298 @@
+                            switch (source[sourceIndex])
+                            {
+								case Utf8Constants.TimePrefix:
+									state = Utf8Constants.DateJParserState.Hour;
+									sourceIndex++;
+									break;
+								case Utf8Constants.UtcOffsetChar:
+								case Utf8Constants.Plus:
+								case Utf8Constants.Minus:
+                                    state = Utf8Constants.DateJParserState.Invalid;
+									break;
+								default:
+									continueParsing = false;
+									break;
+							}
+                        }
+						break;
+					case Utf8Constants.DateJParserState.Hour:
+						state = ParserHelpers.TryGetNextTwoDigits(source, sourceIndex, out sourceIndex, out hour) && (sourceLength - sourceIndex > 1) && (source[sourceIndex++] == Utf8Constants.Colon) ?
+                            Utf8Constants.DateJParserState.Minute:
+                            Utf8Constants.DateJParserState.Invalid;
+						break;
+					case Utf8Constants.DateJParserState.Minute:
+						if (!ParserHelpers.TryGetNextTwoDigits(source, sourceIndex, out sourceIndex, out minute))
+                        {
+                            state = Utf8Constants.DateJParserState.Invalid;
+                            break;
+                        }
+
+                        if (sourceLength - sourceIndex > 1)
+                        {
+                            if (source[sourceIndex] == Utf8Constants.Colon)
+                                state = Utf8Constants.DateJParserState.Second;
+                            else if (source[sourceIndex] == Utf8Constants.UtcOffsetChar)
+				            {
+					            offsetChar = Utf8Constants.UtcOffsetChar;
+					            continueParsing = false;
+				            }
+                            else if (source[sourceIndex] == Utf8Constants.Plus)
+                            {
+                                offsetChar = Utf8Constants.Plus;
+                                state = Utf8Constants.DateJParserState.OffsetHours;
+                            }
+                            else if (source[sourceIndex] == Utf8Constants.Minus)
+                            {
+                                offsetChar = Utf8Constants.Minus;
+                                state = Utf8Constants.DateJParserState.OffsetHours;
+                            }
+                            else
+							{
+								continueParsing = false;
+								break;
+							}
+
+                            sourceIndex++;
+                        }
+						break;
+					case Utf8Constants.DateJParserState.Second:
+						if (!ParserHelpers.TryGetNextTwoDigits(source, sourceIndex, out sourceIndex, out second))
+                        {
+                            state = Utf8Constants.DateJParserState.Invalid;
+                            break;
+                        }
+
+						if (sourceLength - sourceIndex > 1)
+                        {
+                            if (source[sourceIndex] == Utf8Constants.Period)
```

Comment:
inconsistent braces - for nested scopes like this, add braces even for single line branches